plugins/lookup/onepassword.py

```diff
@@ -495,28 +514,44 @@ def assert_logged_in(self):
         return False
 
     def full_signin(self):
-        required_params = [
-            "subdomain",
-            "username",
-            "secret_key",
-            "master_password",
-        ]
-        self._check_required_params(required_params)
+        if self.service_account_token:
+            self._check_required_params(['service_account_token'])
+            environment_update = {"OP_SERVICE_ACCOUNT_TOKEN": self.service_account_token}
+            args = [
+                "whoami",
+            ]
 
-        args = [
-            "account", "add", "--raw",
-            "--address", "{0}.{1}".format(self.subdomain, self.domain),
-            "--email", to_bytes(self.username),
-            "--signin",
-        ]
+            return self._run(args, environment_update=environment_update)
+        else:
+            required_params = [
+                "subdomain",
+                "username",
+                "secret_key",
+                "master_password",
+            ]
+            self._check_required_params(required_params)
 
-        environment_update = {"OP_SECRET_KEY": self.secret_key}
-        return self._run(args, command_input=to_bytes(self.master_password), environment_update=environment_update)
+            args = [
+                "account", "add", "--raw",
+                "--address", "{0}.{1}".format(self.subdomain, self.domain),
+                "--email", to_bytes(self.username),
+                "--signin",
+            ]
+
+            environment_update = {"OP_SECRET_KEY": self.secret_key}
+
+            return self._run(args, command_input=to_bytes(self.master_password), environment_update=environment_update)
 
     def get_raw(self, item_id, vault=None, token=None):
         args = ["item", "get", item_id, "--format", "json"]
         if vault is not None:
             args += ["--vault={0}".format(vault)]
+        if not self.service_account_token and token is not None:
+            args += [to_bytes("--session=") + token]
+        if self.service_account_token:
+            environment_update = {"OP_SERVICE_ACCOUNT_TOKEN": self.service_account_token}
+            return self._run(args, environment_update=environment_update)
```

Comment:
No need to try and add the `--token` arg if there is a service account token (token will be `None` anyway if a valid `service_account_token` was passed to the lookup)

In my testing, I found that a vault is required when using a service account token.
```suggestion

        if self.service_account_token:
            self._check_required_params(["vault"])
            environment_update = {"OP_SERVICE_ACCOUNT_TOKEN": self.service_account_token}
            return self._run(args, environment_update=environment_update)
```